### PR TITLE
Add Linux Support

### PR DESCRIPTION
**Pull request by jpsim**

### CODE REVIEW

**Comment by marcelofabri** [comment]
Maybe we should add a note about testing on Linux on https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md#rules?

**Comment by marcelofabri** [comment]
@jpsim I think we should add an explanation about the `allTests` and `XCTMain` stuff, as people will probably forget to update those as it's not usual on a mac OS environment.

**Comment by marcelofabri** [comment]
This is awesome BTW 💯🎉😍

File: Source/SwiftLintFramework/Rules/RuleConfigurations/PrivateUnitTestConfiguration.swift [hunk 1]
```diff
@@ -13,8 +13,8 @@ public struct PrivateUnitTestConfiguration: RuleConfiguration, Equatable {
     public let identifier: String
     public var name: String?
     public var message = "Regex matched."
-    public var regex = NSRegularExpression()
-    public var included = NSRegularExpression()
+    public var regex = NSRegularExpression.forcePattern("placeholder")
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/RuleConfigurations/PrivateUnitTestConfiguration.swift]: maybe make them optional? 😬

Comment (jpsim) [inline Source/SwiftLintFramework/Rules/RuleConfigurations/PrivateUnitTestConfiguration.swift]: Seems obvious, but I couldn't figure it out earlier. Fixed now.

**Comment by jpsim** [comment]
YOLO

**Pull request merged**